programs/uxd/src/utils/calculate_credix_lp_depository_target_amount.rs

```diff
@@ -0,0 +1,41 @@
+use crate::state::controller::Controller;
+use crate::state::credix_lp_depository::CredixLpDepository;
+use crate::state::identity_depository::IdentityDepository;
+use crate::state::mercurial_vault_depository::MercurialVaultDepository;
+use crate::utils::DepositoryInfoForTargetRedeemableAmount;
+use anchor_lang::prelude::*;
+
+use super::calculate_depositories_target_redeemable_amount;
+
+pub fn calculate_credix_lp_depository_target_amount(
+    controller: &AccountLoader<Controller>,
+    identity_depository: &AccountLoader<IdentityDepository>,
+    mercurial_vault_depository: &AccountLoader<MercurialVaultDepository>,
+    credix_lp_depository: &AccountLoader<CredixLpDepository>,
+) -> Result<u64> {
+    let depositories_target_redeemable_amount = calculate_depositories_target_redeemable_amount(
+        controller.load()?.redeemable_circulating_supply,
+        &vec![
+            // credix is the first in the list
+            DepositoryInfoForTargetRedeemableAmount {
+                weight_bps: controller.load()?.credix_lp_depository_weight_bps,
+                redeemable_amount_under_management_cap: credix_lp_depository
+                    .load()?
+                    .redeemable_amount_under_management_cap,
+            },
+            DepositoryInfoForTargetRedeemableAmount {
+                weight_bps: controller.load()?.identity_depository_weight_bps,
+                redeemable_amount_under_management_cap: identity_depository
+                    .load()?
+                    .redeemable_amount_under_management_cap,
+            },
+            DepositoryInfoForTargetRedeemableAmount {
+                weight_bps: controller.load()?.mercurial_vault_depository_weight_bps,
+                redeemable_amount_under_management_cap: mercurial_vault_depository
+                    .load()?
+                    .redeemable_amount_under_management_cap,
+            },
+        ],
+    )?;
+    Ok(depositories_target_redeemable_amount[0]) // credix is the first in the list
+}
```

Comment:
I was thinking I could also create a "calculate_all_depositories_target_amount" function instead that return a string and use that for the router mint/redeem too, but I wanted to keep things as small as possible here.

That function definitly feels like a hack tho 🤔